extension/browserUtil.js

```diff
@@ -33,6 +33,28 @@ this.browserUtil = (function() {
     });
   };
 
+  exports.activateTab = async function activateTab(url) {
+    if (!url.includes("://")) {
+      url = browser.runtime.getURL(url);
+    }
+    for (const tab of await browser.tabs.query({
+      url: [url]
+    })) {
+      browserUtil.makeTabActive(tab);
+      return;
+    }
+    browser.tabs.create({
```

Comment:
both browserUtil and browser.tabs calls should return (both to give the caller the tab object if they want, and to send back errors if they have an exception)